Source/WarpX.cpp

```diff
@@ -874,10 +874,19 @@ WarpX::ReadParameters ()
         for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
             WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
                 !(
-                    ( WarpX::field_boundary_lo[idim] == FieldBoundaryType::PML &&
-                    WarpX::field_boundary_lo[idim] == FieldBoundaryType::Absorbing_SilverMueller ) ||
-                    ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::PML &&
-                     WarpX::field_boundary_hi[idim] == FieldBoundaryType::Absorbing_SilverMueller )
+                    (std::any_of(WarpX::field_boundary_lo.begin(), WarpX::field_boundary_lo.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::PML;})
+                    ||
+                    std::any_of(WarpX::field_boundary_hi.begin(), WarpX::field_boundary_hi.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::PML;})
+                    )
+                    &&
+                    (std::any_of(WarpX::field_boundary_lo.begin(), WarpX::field_boundary_lo.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::Absorbing_SilverMueller;})
+                    ||
+                    std::any_of(WarpX::field_boundary_hi.begin(), WarpX::field_boundary_hi.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::Absorbing_SilverMueller;})
+                    )
```

Comment:
Actually, I think this assert block now can be placed outside the loop over `idim`, right? Logically it does not depend on `idim` anymore, thanks to the usage of `std::any_of`, if I understand correctly.

Reply:
Ah! Right! You are absolutely right! I've refactored the code.